pallets/runtime/src/multisig.rs

```diff
@@ -272,8 +265,42 @@ decl_module! {
                 <NumberOfSigners<T>>::get(&sender) >= sigs_required,
                 Error::<T>::NotEnoughSigners
             );
-            <MultiSigSignsRequired<T>>::insert(&sender, &sigs_required);
-            Self::deposit_event(RawEvent::MultiSigSignaturesRequiredChanged(sender, sigs_required));
+            Self::unsafe_change_sigs_required(sender, sigs_required);
+            Ok(())
+        }
+
+        /// This function allows to replace all existing signers of the given multisig & also change no. of signature required
+        /// NOTE - Once this function get executed no other function of the multisig is allowed to execute until unless
+        /// potential signers accept the authorisation and there count should be greater than or equal to the signature required
+        ///
+        /// # Arguments
+        /// * signers - Vector of signers for a given multisig
+        /// * sigs_required - Number of signature required for a given multisig
+        pub fn change_all_signers_and_sigs_required(origin, signers: Vec<Signatory>, sigs_required: u64) -> DispatchResult {
+            let sender = ensure_signed(origin)?;
+            let sender_signer = Signatory::from(AccountKey::try_from(sender.encode())?);
+            ensure!(<MultiSigSignsRequired<T>>::exists(&sender), Error::<T>::NoSuchMultisig);
+            ensure!(signers.len() > 0, Error::<T>::NoSigners);
+            ensure!(u64::try_from(signers.len()).unwrap_or_default() >= sigs_required && sigs_required > 0,
+                Error::<T>::RequiredSignaturesOutOfBounds
+            );
+
+            // Removing the signers from the valid multi-signers list first
+            let old_signers = <MultiSigSigners<T>>::iter_prefix(&sender).collect::<Vec<Signatory>>();
+            old_signers.into_iter()
+                .for_each(|signer| {
+                    Self::unsafe_signer_removal(sender.clone(), signer)
+                });
+
+            // Add the new signers for the given multi-sig
+            signers.into_iter()
+                .for_each(|signer| {
+                    Self::unsafe_add_auth_for_signers(sender_signer, signer, sender.clone())
+                });
```

Comment:
We can optimize this by filtering out the common signers between new and old and only calling these functions for the remaining signer.